Explorer/Assets/DCL/MapRenderer/MapLayers/LiveEvents/LiveEventsMarkersController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.EventsApi;
+using DCL.MapRenderer.Culling;
+using DCL.PlacesAPIService;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Pool;
+using ICoordsUtils = DCL.MapRenderer.CoordsUtils.ICoordsUtils;
+
+namespace DCL.MapRenderer.MapLayers.Categories
+{
+    internal class LiveEventsMarkersController : MapLayerControllerBase, IMapCullingListener<ICategoryMarker>, IMapLayerController, IZoomScalingLayer
+    {
+        private static readonly TimeSpan LIVE_EVENTS_POLLING_TIME = TimeSpan.FromMinutes(5);
+        private readonly MapLayer mapLayer;
+
+        internal delegate ICategoryMarker CategoryMarkerBuilder(
+            IObjectPool<CategoryMarkerObject> objectsPool,
+            IMapCullingController cullingController,
+            ICoordsUtils coordsUtils);
+
+        private readonly IObjectPool<CategoryMarkerObject> objectsPool;
+        private readonly CategoryMarkerBuilder builder;
+        private readonly CategoryIconMappingsSO categoryIconMappings;
+        private readonly IEventsApiService eventsApiService;
+        private readonly ClusterController clusterController;
+
+        private readonly Dictionary<Vector2Int, IClusterableMarker> markers = new();
+
+        private Vector2Int decodePointer;
+        private bool isEnabled;
+        private int zoomLevel = 1;
+        private float baseZoom = 1;
+        private float zoom = 1;
+        private bool arePlacesLoaded;
+
+        public LiveEventsMarkersController(
+            IEventsApiService eventsApiService,
+            IObjectPool<CategoryMarkerObject> objectsPool,
+            CategoryMarkerBuilder builder,
+            Transform instantiationParent,
+            ICoordsUtils coordsUtils,
+            IMapCullingController cullingController,
+            CategoryIconMappingsSO categoryIconMappings,
+            MapLayer mapLayer,
+            ClusterController clusterController)
+            : base(instantiationParent, coordsUtils, cullingController)
+        {
+            this.eventsApiService = eventsApiService;
+            this.objectsPool = objectsPool;
+            this.builder = builder;
+            this.categoryIconMappings = categoryIconMappings;
+            this.mapLayer = mapLayer;
+            this.clusterController = clusterController;
+        }
+
+        public UniTask InitializeAsync(CancellationToken cancellationToken)
+        {
+            LoadEventsPlacesAsync(cancellationToken).Forget();
+            return UniTask.CompletedTask;
+        }
+
+        private async UniTaskVoid LoadEventsPlacesAsync(CancellationToken ct)
```

Comment:
Is this polling called only when the layer is visible and the map is opened, right?

Reply:
no, because they are visible in the minimap as well, so one request every 5 mins no matter ehat